windows/devices/uwb/UwbDeviceConnector.cxx

```diff
@@ -47,17 +47,34 @@ UwbDeviceConnector::Reset()
         return resultFuture;
     }
 
-    UWB_STATUS status;
-    BOOL ioResult = DeviceIoControl(handleDriver.get(), IOCTL_UWB_DEVICE_RESET, nullptr, 0, &status, sizeof status, nullptr, nullptr);
+    ULONG deviceResetInputBufferLength = sizeof(UWB_DEVICE_RESET);
+    std::vector<uint8_t> deviceResetInputBuffer(deviceResetInputBufferLength);
+    auto reset = reinterpret_cast<UWB_DEVICE_RESET*>(std::data(deviceResetInputBuffer));
+
+    reset->size = deviceResetInputBufferLength;
+    reset->resetConfig = UWB_RESET_CONFIG_UWBS_RESET;
+
+    std::vector<uint8_t> deviceResetStatusBuffer{};
+    DWORD bytesRequired = sizeof(UWB_STATUS);
+    deviceResetStatusBuffer.resize(bytesRequired);
+
+    BOOL ioResult = DeviceIoControl(handleDriver.get(), IOCTL_UWB_DEVICE_RESET, std::data(deviceResetInputBuffer), deviceResetInputBufferLength, std::data(deviceResetStatusBuffer), std::size(deviceResetStatusBuffer), &bytesRequired, nullptr);

```

Comment:
Neither the `vector` for either arg nor the `bytesRequired` arg should be required here since the sizes of both args are fixed. Can this work?

```suggestion
    const UWB_DEVICE_RESET deviceReset {
        .size = sizeof(UWB_DEVICE_RESET),
        .resetConfig = UWB_RESET_CONFIG_UWBS_RESET,
    };
    UWB_STATUS status;
    BOOL ioResult = DeviceIoControl(handleDriver.get(), IOCTL_UWB_DEVICE_RESET, const_cast<UWB_DEVICE_RESET *>(&deviceReset), sizeof deviceReset, &status, sizeof status, nullptr, nullptr);

```

Reply:
Ah I thought bytesRequired was necessary for DeviceIoControl, but that's only if the "overlapped" field is not null, which in this case it is. Your suggested change works so I will push the changes